docs/Advanced-Topics.rst

```diff
@@ -77,3 +77,26 @@ Recommendations for gcc Users (MinGW, \*nix)
+
+::
+
+    3
+    4
+    1
+    ...
+
+It means the position of the document from the first data row, when presented to a user, was ``3``, second was ``4``, and so on. A position could be any (arbitrary) integer number and could encode the presentation of an item even in complex layouts (it is up to the LightGBM user to perform such encoding/enumeration).
+
+The position file corresponds with training data file line by line, and has one position per line.
+
+And if the name of training data file is ``train.txt``, the position file should be named as ``train.txt.position`` and placed in the same folder as the data file.
+In this case, LightGBM will load the position file automatically if it exists.
+
+Also, you can include position column in your data file. Please refer to the ``position_column`` `parameter <#position_column>`__ in above.
+
+Currently, implemented is an approach to model position bias by using an idea of Generalized Additive Models (`GAM <https://en.wikipedia.org/wiki/Generalized_additive_model>`_) to linearly decompose the document score ``s`` into the sum of a relevance component ``f`` and a positional component ``g``:  ``s(x, pos) = f(x) + g(pos)`` where the former component depends on the original query-document features and the latter depends on the position of an item. During the training, the compound scoring function ``s(x, pos)`` is fit with a standard ranking algorithm (e.g., LambdaMART) which boils down to jointly learning the relevance component ``f(x)`` (it is later returned as an unbiased model) and the position factors ``g(pos)`` that help better explain the observed (biased) labels. Similar score decomposition ideas have previously been applied for classification & pointwise ranking tasks with assumptions of binary labels and binary relevance (a.k.a. "two-tower" models, refer to the papers: `Towards Disentangling Relevance and Bias in Unbiased Learning to Rank <https://arxiv.org/abs/2212.13937>`_, `PAL: a position-bias aware learning framework for CTR prediction in live recommender systems <https://dl.acm.org/doi/10.1145/3298689.3347033>`_, `A General Framework for Debiasing in CTR Prediction <https://arxiv.org/abs/2112.02767>`_). In LightGBM, we adapt this idea to general pairwise Lerarning-to-Rank with arbitrary ordinal relevance labels. Besides, GAMs have been used in the context of explainable ML (`Accurate Intelligible Models with Pairwise Interactions <https://www.cs.cornell.edu/~yinlou/papers/lou-kdd13.pdf>`_) to linearly decompose the contribution of each feature (and possibly their pairwise interactions) to the overall score, for subsequent analysis and interpretation of their effects in the trained models.
```

Comment:
For every line in this file, and especially this line, please break them up into one sentence per line.

Consecutive lines not separated by a blank line will be correctly combined back together into a paragraph with wrapping, so the visual appearance will be identical.

But having each sentence on its own line makes it easier to review changes to the doc, and makes the git blame view more useful as a source of "when was this last changed, why, and by who?".